src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/SPARQLStar2CypherTranslatorImpl.java

```diff
@@ -16,14 +16,26 @@
 import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.query.impl.match.NodeMatchClause;
 import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.query.impl.returns.*;
 import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.utils.CypherQueryBuilder;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.utils.CypherVarGenerator;
 
 import java.util.*;
 
 public class SPARQLStar2CypherTranslatorImpl implements SPARQLStar2CypherTranslator {
 
+    protected final CypherVarGenerator generator;
+
+    public SPARQLStar2CypherTranslatorImpl() {
+        this.generator = new CypherVarGenerator();
+    }
+
     @Override
     public CypherQuery translateTriplePattern(final TriplePattern tp, final LPG2RDFConfiguration conf) {
-        return translateTriplePattern(tp, conf, new CypherVarGenerator(), new HashSet<>());
+        return translateTriplePattern(tp, conf, this.generator, new HashSet<>());
```

Comment:
I would prefer this class to remain stateless. The new implementation with the `generator` member variable is problematic because, for every translation, you need to create a new object of this class because you need a fresh `CypherVarGenerator`. Hence, this prevents you from being able to reuse an object of this class for many translations, which is not what I would assume from looking just at the implemented interface (`SPARQLStar2CypherTranslator`).

Here is how you can avoid this issue: revert this class back to how it was before and, then, change the return type of this method (in the interface) to be `Pair<CypherQuery,CypherVarGenerator>`. That way, you can create a new `CypherVarGenerator` for every translation (exactly as it was done before). In fact, you may even change the return type to be `Pair<CypherQuery, Map<CypherVar,Node>>` where this `Map<CypherVar,Node>` is populated inside the `CypherVarGenerator` and, in the end, can be obtained from the `CypherVarGenerator` and returned here.

This `Map<CypherVar,Node>` can then be passed to the methods of `Record2SolutionMappingTranslator` (instead of passing the whole `CypherVarGenerator`).